Update keyboard.md to include returned object on keyboardDidShow and keyboardDidHide events (#3431)
facebook/react-native-website: docs/keyboard.md

```

Useful for syncing TextInput (or other keyboard accessory view) size of position changes with keyboard movements.

## Type Definitions

### KeyboardMetrics

> `KeyboardMetrics` object is part of [`Keyboard`](text#addListener) callback and contains the measurement data of `Keyboard`.

Reviewer: We also use this in a static function `Keyboard.metrics()` (that I forgot to add to documentation when I added the API...). Instead of linking to the list of APIs using this, could we instead link to this from that API?

I.e. add a reference to `KeyboardMetrics` in the `addListener` section, but replace the KeyboardMetrics description with a description of what the type is?

```suggestion
`KeyboardMetrics` represents the size and position of an on-screen keyboard.
```

`on-screen` and `virtual` usually mean the same thing, except for VR which can have a soft keyboard outside of screen-space. So the metrics are *technically* the intersection of the keyboard and the screen. But I don't know how much detail we need about that case.

#### Example

```js
{
  screenX: 0,
  screenY: 401.1428527832031,
  width: 411.4285583496094,
  height: 282.28570556640625,
}
```

#### Properties

| Name      | Type    | Optional  | Description                                                         |
| --------- | ------- | --------- | ------------------------------------------------------------------- |

> | screenX   | number  | No        | Virtual keyboard X position on screen in pixels.                    |

Reviewer: Might reduce ambiguity to specify that we are referring to device-independent pixels instead of screen pixels.

```suggestion
| screenX   | number  | No        | Virtual keyboard X position on screen in CSS pixels.                    |
```

| screenY   | number  | No        | Virtual keyboard Y position on screen in pixels.                    |
| width     | number  | No        | Virtual keyboard width in pixels.                                   |
| height    | number  | No        | Virtual keyboard height in pixels.                                  |

### KeyboardEvent

> `KeyboardEvent` object is returned in the callback as a result of a [`Keyboard`](text#addListener) event. It contains a key called `endCoordinates` with a value which is an object of type [`KeyboardMetrics`](text#KeyboardMetrics).

Reviewer: Could we link to KeyboardEvent in `addListener()`?

#### Example

```js
{
  duration: 0,
  easing: "keyboard",
  endCoordinates: KeyboardMetrics
}
```

#### Properties

| Name           | Type                                                                       | Optional | Description                                    |
| -------------- | -------------------------------------------------------------------------- | -------- | ---------------------------------------------- |

> | duration       | number                                                                     | No       | Virtual keyboard animation duration.           |
> | easing         | enum(`"easeIn"`, `"easeInEaseOut"`, `"easeOut"`, `"linear"`, `"keyboard"`) | No       | Virtual keyboard animation easing function.    |
> | endCoordinates | [KeyboardMetrics](text#KeyboardMetrics) object                             | No       | Object containing `Keyboard` measurement data. |

Reviewer 2: ```suggestion
| duration       | number                                                                     | No       | Virtual keyboard animation duration.           |
| easing         | enum(`"easeIn"`, `"easeInEaseOut"`, `"easeOut"`, `"linear"`, `"keyboard"`) | No       | Virtual keyboard animation easing function.    |
| endCoordinates | [KeyboardMetrics](text#KeyboardMetrics) object                             | No       | Object containing `Keyboard` measurement data. |
```

Few things in this table:
* we are missing iOS only fields - ` startCoordinates` and `isEventFromThisApp`
* `easing` is always `'keyboard'` on Android
* `duration` is always `0` on Android

You can see the example of marking platform specific fields in type tables here:
* https://reactnative.dev/docs/next/image#imagesource
